doc/lsst.pipe.base/creating-a-task.rst

```diff
@@ -0,0 +1,350 @@
+.. NOTE: this topic should be edited into the modernized topic-based documentation style.
+
+.. _creating-a-task:
+
+###############
+Creating a task
+###############
+
+This document describes how to write a data processing task.
+For an introduction to data processing tasks, please read :doc:`task-framework-overview`.
+It also helps to have a basic understanding of data repositories and how to use the butler to read and write data (to be written; for now read existing tasks to see how it is done).
+
+After reading this document you may wish to read :doc:`creating-a-command-line-task`.
+
+.. _creating-a-task-intro:
+
+Introduction
+============
+
+Tasks are subclasses of `lsst.pipe.base.Task` in general or `lsst.pipe.base.CmdLineTask` for command-line tasks.
+
+Tasks are constructed by calling ``__init__`` with the :ref:`task configuration <creating-a-task-configuration>`.
+Occasionally additional arguments are required (see the task's documentation for details).
+`lsst.pipe.base.Task` has a few other arguments that are usually only specified when a task is created as a subtask of another task; you will probably never have to specify them yourself.
+(Command-line tasks are often constructed and run by calling `lsst.pipe.base.parseAndRun`, as described in :doc:`creating-a-command-line-task`.)
+
+Tasks typically have a ``run`` method that executes the task's main function.
+See :ref:`creating-a-task-class-methods` for more information.
+
+.. _creating-a-task-configuration:
+
+Configuration
+=============
+
+Every task requires a configuration: a task-specific set of configuration parameters.
+The configuration is read-only; once you construct a task, the same configuration will be used to process all data with that task.
+This makes the data processing more predictable: it does not depend on the order in which items of data are processed.
```

Comment:
The order of configuration of multiple tasks is also critical, as e.g. tasks can add to an `afwTask.schema` as they are configured;  all of this must be done before the first task is processed.  This provides another reason why the Tasks are first created and configured, and only then run.